units/patch_intent.go

```diff
@@ -527,29 +527,40 @@ func (j *patchIntentProcessor) setToPreviousPatchDefinition(patchDoc *patch.Patc
 		if err = setTasksToPreviousFailed(patchDoc, previousPatch, project); err != nil {
 			return "", errors.Wrap(err, "settings tasks to previous failed")
 		}
-
 	} else {
 		patchDoc.Tasks = previousPatch.Tasks
 	}
 
 	return previousPatch.Status, nil
 }
 
-func getPreviousFailedTasksAndDisplayTasks(tasksInProjectVariant []string, vt patch.VariantTasks, version string) ([]string, error) {
+func getPreviousFailedTasksAndDisplayTasks(project *model.Project, tasksInProjectVariant []string, vt patch.VariantTasks, version string) ([]string, error) {
 	failedTasks, err := task.FindAll(db.Query(task.FailedTasksByVersionAndBV(version, vt.Variant)))
 	if err != nil {
 		return nil, errors.Wrapf(err, "finding failed tasks in build variant '%s' from previous patch '%s'", vt.Variant, version)
 	}
 	failedExecutionTasks := []string{}
+	taskGroupFailedTasks := map[string][]string{}
```

Comment:
Why do we store these in a separate place rather than just adding them to failedExecutionTasks? They would hit the same StringSliceIntersection which would remove any deduplication (I'm thinking it might be to check that g is in tasksInProjectVariant but I think now that you're passing in the current project you're already handling that just by using project.FindTaskGroup)

Reply:
updated as we spoke about in meeting!